File: runtime/python/units/rt/i64_div_s.py
# requires: rt/s64, rt/trap
@staticmethod
def i64_div_s(a, b):
    sa = Rt.s64(a)
    sb = Rt.s64(b)
    if sb == 0:
        Rt.trap("integer divide by zero")
    q = abs(sa) // abs(sb)
    if (sa < 0) ^ (sb < 0):
        q = -q
    if q > 0x7FFFFFFFFFFFFFFF:
        Rt.trap("integer overflow")
    return q & Rt.M64

File: runtime/ruby/units/rt/i64_div_s.rb
# requires: rt/s64, rt/trap, rt/m64
def i64_div_s(a, b)
  sa = s64(a)
  sb = s64(b)
  trap("integer divide by zero") if sb == 0
  q = sa.abs / sb.abs
  q = -q if (sa < 0) ^ (sb < 0)
  trap("integer overflow") if q > 0x7fff_ffff_ffff_ffff
  m64(q)
end
